src/EspressoSGXTEEVerifier.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {V3QuoteVerifier} from
+    "@automata-network/dcap-attestation/contracts/verifiers/V3QuoteVerifier.sol";
+import {BELE} from "@automata-network/dcap-attestation/contracts/utils/BELE.sol";
+import {Header} from "@automata-network/dcap-attestation/contracts/types/CommonStruct.sol";
+import {
+    HEADER_LENGTH,
+    ENCLAVE_REPORT_LENGTH
+} from "@automata-network/dcap-attestation/contracts/types/Constants.sol";
+import {EnclaveReport} from "@automata-network/dcap-attestation/contracts/types/V3Structs.sol";
+import {BytesUtils} from "@automata-network/dcap-attestation/contracts/utils/BytesUtils.sol";
+import "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {IEspressoSGXTEEVerifier} from "./interface/IEspressoSGXTEEVerifier.sol";
+
+/**
+ *
+ * @title  Verifies quotes from the TEE and attests on-chain
+ * @notice Contains the logic to verify a quote from the TEE and attest on-chain. It uses the V3QuoteVerifier contract
+ *         from automata to verify the quote. Along with some additional verification logic.
+ */
+contract EspressoSGXTEEVerifier is IEspressoSGXTEEVerifier, Ownable2Step {
+    using BytesUtils for bytes;
+
+    // V3QuoteVerififer contract from automata to verify the quote
+    V3QuoteVerifier public quoteVerifier;
+
+    mapping(bytes32 => bool) public registeredEnclaveHash;
+    mapping(bytes32 => bool) public registeredEnclaveSigner;
+    mapping(address => bool) public registeredSigners;
+
+    constructor(bytes32 enclaveHash, bytes32 enclaveSigner, address _quoteVerifier)
+        Ownable(msg.sender)
+    {
+        quoteVerifier = V3QuoteVerifier(_quoteVerifier);
```

Comment:
you can add checks to ensure that the values are valid e.g. 
```solidity
if (_quoteVerifier == address(0)) return InvalidValue()
```

Also is the quoteVerifier meant to be a contract? If so, you can also use `_quoteVerifier.code.length > 0` to determine if it is a contract but this assumes that this contract is already deployed when this contract, `EspressoSGXTEEVerifier` is being deployed.

Reply:
https://github.com/EspressoSystems/espresso-tee-contracts/pull/2/commits/a2bf9e83c0441bd275e4cb10f5bb23b8d440a8ad